From 4c1c0c86df3a6e857bacfc5abf02a44bbe10d8bf Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Sat, 15 Feb 2014 13:46:59 -0800
Subject: [PATCH 541/974] MIPS: OCTEON: Fix io_resource end value

Their are only 2 bits for pcie port, for 4 ports the pcie ports are 0..3.

Also fixed io_map_base, it is same for all PCIe interfaces.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/pci/pcie-octeon.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/mips/pci/pcie-octeon.c b/arch/mips/pci/pcie-octeon.c
index a02befe..f22260c 100644
--- a/arch/mips/pci/pcie-octeon.c
+++ b/arch/mips/pci/pcie-octeon.c
@@ -566,7 +566,7 @@ static int __init octeon_pcie_setup(void)
 	set_io_port_base(CVMX_ADD_IO_SEG(cvmx_pcie_get_io_base_address(0)));
 	ioport_resource.start = 0;
 	ioport_resource.end =
-		cvmx_pcie_get_io_base_address(CVMX_PCIE_PORTS) -
+		cvmx_pcie_get_io_base_address(CVMX_PCIE_PORTS-1) -
 		cvmx_pcie_get_io_base_address(0) + cvmx_pcie_get_io_size(1) - 1;
 
 	/*
@@ -634,6 +634,7 @@ static int __init octeon_pcie_setup(void)
 			case 0:
 			case 1:
 			case 2:
+			case 3:
 				octeon_pcie_interface_init(&octeon_pcie[port], 0, port);
 				/* Memory offsets are physical addresses */
 				octeon_pcie[port].controller.mem_offset = cvmx_pcie_get_mem_base_address(port);
@@ -646,7 +647,7 @@ static int __init octeon_pcie_setup(void)
 				 * based on first slot's value so that both the routines will
 				 * work properly.
 				 */
-				octeon_pcie[port].controller.io_map_base = CVMX_ADD_IO_SEG(cvmx_pcie_get_io_base_address(port));
+				octeon_pcie[port].controller.io_map_base = CVMX_ADD_IO_SEG(cvmx_pcie_get_io_base_address(0));
 				/*
 				 * To keep things similar to PCI, we start
 				 * device addresses at the same place as PCI
-- 
2.6.2

